src/libtiled/wangset.cpp

```diff
@@ -134,6 +187,71 @@ void WangId::flipVertically()
     rotate(2);
 }
 
+WangIdVariationIterator::WangIdVariationIterator(int edgeColors, int cornerColors, WangId wangId)
```

Comment:
The point of it is to make "WildWangIdUsed" a lot faster. That function is pretty important, so Im still working on speeding it up more.

In regards to the iterator itself, I know it isn't proper with end being valid, but the problem is in the 15 x 15 case, all unsigned ints are valid, so I'm not sure there is a way to make it invalid to dereference. So I'm not sure I can make it proper? Though I see your point it would be nice to avoid the allocation... Though currently, the variations function is unused (it is in the findMatchingWangTiles function, but that can be changed to use the iterator as is), so It could be removed all together. Ill work on a more elegant solution though,